test/Autofac.Specification.Test/Autofac.Specification.Test.csproj

```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFrameworks>net5.0</TargetFrameworks>
```

Comment:
Is there potentially value in multi-targeting the test projects so they run under the different frameworks? We did this for a bit when it was .NET 4.5 and netstandard because we had conditional tests. Now we have the shims that are conditional... but, then, the .NET 4.5 stuff was about partial trust and code access security crap, so maybe it doesn't make sense to repeat the same tests twice.

Reply:
It also means our build system would need to change (which is why I switched back to a single target); currently we only install one version of the .NET SDK; we'd need to install two.

Thinking about it though, I did explicitly `#if` out a block of DependencyInjectionException tests to run on .NET5 (to do with `BinaryFormatter`, but now that test just won't run at all.

Do we need to amend our build system to install netcoreapp3.1 & net5.0 versions?

Reply:
Yeah, I think we probably have to get that sorted. Once we start doing `#if` stuff to support different frameworks it seems like we should test both code paths out. It's one of the big reasons I hate `#if` conditional compilation; it makes everything harder to test and maintain. I was super glad to get rid of the code access security stuff because of that.

Reply:
OK; I've added a new option into the global.json (seemed like the best place for it) that specifies an older version of the SDK to install in addition to the latest. Tests are now running on both frameworks.

Reply:
Just as a hint; When you do multi-target, then there are two different code-coverage files generated. Both have to be uploaded then AFAIK.

Reply:
Thanks @alsami; fixing the coverage was something of a pain because of the multiple files as you said. If you had a few moments to check my changes are ok, that would be appreciated.

Basically had to end up putting the coverage artifacts in separate folders, because the `MergeWith` property can't be framework-specific.